### PR TITLE
Remove special handling for bootstrap period - Closes #5780

### LINKED ISSUE #5780: DPoS should not ban or accept punishment during bootstrap period
### Description
- Add logic in `lisk-chain/setValidators` to not to update the validator during the bootstrap period.
- Remove bootstrap specific logic from dpos afterBlockApply (it should calculate productivity during bootstrap period)

### Expected behavior
- During the bootstrap period, all productivity calculation should happen
- Validator should not be changed during the bootstrap period

### Actual behavior
- DPoS is not calculating the productivity during the bootstrap period
- lisk-chain does not protect the validator change in function

### Which version(s) does this affect? (Environment, OS, etc...)
5.0-

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #5780 

### How was it solved?

- Add logic in setValidator not to update validator during bootstrap period
- Remove special handling of bootstrap period from dpos module

### How was it tested?

- Add unit test case to check if validator is not updated during bootstrap period

### CODE REVIEW

**Review by MaximeGagnebin — APPROVED** [review]
Looks good.
I have to say that this PR is a bit outside my comfort zone for review.

File: elements/lisk-chain/src/chain.ts [hunk 1]
```diff
@@ -433,7 +426,16 @@ export class Chain {
 	public async setValidators(
 		validators: { address: Buffer; isConsensusParticipant: boolean }[],
 		stateStore: StateStore,
+		blockHeader: BlockHeader,
```
Comment (nazarhussain) [inline elements/lisk-chain/src/chain.ts:429]: I believe the chain should be aware o the last block header and validators should be always updated based on last block. So need to pass extra block header param.

File: framework/src/modules/dpos/dpos_module.ts [hunk 2]
```diff
@@ -175,6 +167,23 @@ export class DPoSModule extends BaseModule {
 		}
 
 		await setRegisteredDelegates(context.stateStore, { registeredDelegates: delegateUsernames });
+
+		const initialRound = this.rounds.calcRound(context.genesisBlock.header.height) + 1;
```
Comment (nazarhussain) [inline framework/src/modules/dpos/dpos_module.ts:171]: Suggest to use some name other than `initialRound`, its got mixed up with the `asset.initRounds` property.

**Review by nazarhussain — CHANGES_REQUESTED** [review]

File: elements/lisk-chain/src/chain.ts (hunk 1 above)
Comment (shuse2) [reply elements/lisk-chain/src/chain.ts:429]: This `blockHeader` is not the `lastBlock` that chain holds, but is currently being processed block.
Therefore, it needs to be passed to this function.

ex)
this.lastBlock.height === 2 then blockHeader.height will be 3

Comment (nazarhussain) [reply elements/lisk-chain/src/chain.ts:429]: If that's the case always, can't we use that offset ourself in chain? to reduce an extra parameter and give user flexibility to update validators in custom module without need to worry about block header. What do you think?

Comment (shuse2) [reply elements/lisk-chain/src/chain.ts:429]: Actually, from the custom module, this parameter is not exposed.
https://github.com/LiskHQ/lisk-sdk/pull/5782/files#diff-b543eea0d99418fe1fb34ff986a2e848R450-R452
Custom module will not need to pass the block header manually

**Review by nazarhussain — APPROVED** [review]

**Pull request merged**